src/System.CommandLine.Tests/Invocation/CommandHandlerTests.cs

```diff
@@ -507,5 +507,101 @@ public class OverridenVirtualTestCommandHandler : VirtualTestCommandHandler
             public override Task<int> InvokeAsync(InvocationContext context)
                 => Task.FromResult(41);
         }
+
+        [Fact]
+        public static void FromBindingContext_forwards_invocation_to_bound_handler_type()
+        {
+            var command = new RootCommand
+            {
+                Handler = CommandHandler.FromBindingContext<BindingContextResolvedCommandHandler>()
+            };
+            command.Handler.Should().NotBeOfType<BindingContextResolvedCommandHandler>();
```

Comment:
This looks like it deserves to be a separate test to surface this detail.

Reply:
resolved in https://github.com/dotnet/command-line-api/pull/1358/commits/5db4df8aa57b46a17b3a60c57c0a92a847fad59b